pkg/client/cache/fifo.go

```diff
@@ -22,12 +22,15 @@ import (
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
+type PopProcessFunc func(interface{}) error
+
 // Queue is exactly like a Store, but has a Pop() method too.
 type Queue interface {
 	Store
 
-	// Pop blocks until it has something to return.
-	Pop() interface{}
+	// Pop blocks until it has something to process.
+	// It returns the object that was process and the result of processing.
+	Pop(PopProcessFunc) (interface{}, error)
```

Comment:
I understand the intention. However, I dislike this interface change. I would still keep the Queue is a simple data structure without the ability to process the out-coming data. Is there anything prevent us from doing this outside the queue?